C++/C++.sublime-syntax

```diff
@@ -205,31 +210,31 @@ contexts:
     # https://en.cppreference.com/w/cpp/language/integer_literal
 
     # hexadecimal integer
-    - match: \b(0[xX])({{hex_digit}}*)({{hex_suffix}})?
+    - match: '{{hex_integer}}'
```

Comment:
By moving those patterns to variables its just harder to keep track of capture groups.

The only reason to introduce variables was to provide an easier way for an inherited syntax to tweak those patterns without touching contexts. 

But I guess that's not scope of this PR.